Storage/src/Connection/Rest.php

```diff
@@ -305,8 +307,10 @@ private function resolveUploadOptions(array $args)
         }
 
         // @todo add support for rolling hash
-        if ($args['validate'] && !isset($args['metadata']['md5Hash'])) {
+        if (($args['validate'] === true || $args['validate'] === 'md5') && !isset($args['metadata']['md5Hash'])) {
             $args['metadata']['md5Hash'] = base64_encode(Psr7\hash($args['data'], 'md5', true));
+        } elseif ($args['validate'] === 'crc32' && !isset($args['metadata']['crc32c'])) {
+            $args['metadata']['crc32c'] = $this->crcFromStream($args['data']);
```

Comment:
@frankyn 
https://github.com/googleapis/google-cloud-php/blob/1d45b6cac16c5457dad786a3ea6446a6e268be1e/Storage/src/Connection/Rest.php#L299

`$args['data']` will always be a StreamInterface.

Reply:
lol no worries. 

Gotcha, Thank you!